cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -20,18 +20,30 @@
 /**
  * Defines a cron-based schedule for running a program. 
  */
-public class Schedule {
+public abstract class Schedule {
+  /**
+   * Schedule Type.
+   */
+  public enum ScheduleType { TIME, DATA }
```

Comment:
But actually we could make this enum part of the ``ScheduleSpecificationCodec`` itself, because it is in the end the only one that really needs this information. In other places, we can use the type of the ``Schedule`` instead of the enum.